app/keepers/keepers.go

```diff
@@ -277,19 +279,51 @@ func NewAppKeeper(
 		govRouter,
 	)
 
+	// IBC Fee Module keeper
+	appKeepers.IBCFeeKeeper = ibcfeekeeper.NewKeeper(
+		appCodec, appKeepers.keys[ibcfeetypes.StoreKey],
+		appKeepers.GetSubspace(ibcfeetypes.ModuleName),
+		appKeepers.IBCKeeper.ChannelKeeper, // todo may be replaced with IBC middleware
+		appKeepers.IBCKeeper.ChannelKeeper,
+		&appKeepers.IBCKeeper.PortKeeper, appKeepers.AccountKeeper, appKeepers.BankKeeper,
+	)
+
 	appKeepers.TransferKeeper = ibctransferkeeper.NewKeeper(
 		appCodec,
 		appKeepers.keys[ibctransfertypes.StoreKey],
 		appKeepers.GetSubspace(ibctransfertypes.ModuleName),
-		appKeepers.IBCKeeper.ChannelKeeper,
+		appKeepers.IBCFeeKeeper,
 		appKeepers.IBCKeeper.ChannelKeeper,
 		&appKeepers.IBCKeeper.PortKeeper,
 		appKeepers.AccountKeeper,
 		appKeepers.BankKeeper,
 		appKeepers.ScopedTransferKeeper,
 	)
 	appKeepers.TransferModule = transfer.NewAppModule(appKeepers.TransferKeeper)
-	transferIBCModule := transfer.NewIBCModule(appKeepers.TransferKeeper)
+
+	appKeepers.RouterKeeper = routerkeeper.NewKeeper(
+		appCodec, appKeepers.keys[routertypes.StoreKey],
+		appKeepers.GetSubspace(routertypes.ModuleName),
+		appKeepers.TransferKeeper,
+		appKeepers.IBCKeeper.ChannelKeeper,
+		appKeepers.DistrKeeper,
+		appKeepers.BankKeeper,
+		&appKeepers.IBCKeeper.PortKeeper, // todo check here
+		appKeepers.IBCFeeKeeper,
+	)
+
+	appKeepers.RouterModule = router.NewAppModule(appKeepers.RouterKeeper)
+
+	var transferStack porttypes.IBCModule
+	transferStack = transfer.NewIBCModule(appKeepers.TransferKeeper)
+	transferStack = ibcfee.NewIBCMiddleware(transferStack, appKeepers.IBCFeeKeeper)
+	transferStack = router.NewIBCMiddleware(
+		transferStack,
+		appKeepers.RouterKeeper,
+		0,
+		routerkeeper.DefaultForwardTransferPacketTimeoutTimestamp,
+		routerkeeper.DefaultRefundTransferPacketTimeoutTimestamp,
+	)
```

Comment:
I think you will need to swap these two lines. 

From what I'm looking at, the desired flow is:
`SendPacket`:  transfer -> router -> fee -> ibc core
`RecvPacket`: ibc core -> fee -> router -> transfer

```suggestion
	var transferStack porttypes.IBCModule
	transferStack = transfer.NewIBCModule(appKeepers.TransferKeeper)
	transferStack = router.NewIBCMiddleware(
		transferStack,
		appKeepers.RouterKeeper,
		0,
		routerkeeper.DefaultForwardTransferPacketTimeoutTimestamp,
		routerkeeper.DefaultRefundTransferPacketTimeoutTimestamp,
	)
	transferStack = ibcfee.NewIBCMiddleware(transferStack, appKeepers.IBCFeeKeeper)
```

What you're seeing currently is ibc core passing an ics29 fee acknowledgement to the packet forward router module. And the router failing to unmarshal the ack [here](https://github.com/strangelove-ventures/packet-forward-middleware/blob/v4.0.0/router/ibc_middleware.go#L254-L257). 

If ibc core passes the acknowledgement firstly to ics29 fee, it should successfully [unmarshal the incentivized ack](https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/ibc_middleware.go#L252-L255) and then [pass `ack.AppAcknowledgement` to the next handler in the call stack](https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/ibc_middleware.go#L292).

Reply:
I see, thank you, Damian!👍